test/src/test/java/org/corfudb/runtime/collections/PersistedCorfuTableTest.java

```diff
@@ -1032,11 +1044,62 @@ Arbitrary<Set<EventInfo>> eventInfoSet() {
         return eventInfo().set();
     }
 
+    @Provide
+    Arbitrary<Set<Person>> personSet() {
+        return person().set();
+    }
+
     /**
      * Check {@link PersistedCorfuTable} integration with {@link CorfuStore}.
      */
     @Property(tries = NUM_OF_TRIES)
     void dataStoreIntegration(
+            @ForAll @StringLength(min = STRING_MIN, max = STRING_MAX) @AlphaChars String namespace,
+            @ForAll @StringLength(min = STRING_MIN, max = STRING_MAX) @AlphaChars String tableName,
+            @ForAll("uuidSet") @Size(SAMPLE_SIZE + 1) Set<Uuid> ids,
+            @ForAll("personSet") @Size(SAMPLE_SIZE + 1) Set<Person> persons) throws Exception {
+        resetTests();
+
+        // Creating Corfu Store using a connected corfu client.
+        CorfuStore corfuStore = new CorfuStore(getDefaultRuntime());
+
+        // Create & Register the table.
+        // This is required to initialize the table for the current corfu client.
+        final Path persistedCacheLocation = Paths.get(diskBackedDirectory, defaultTableName);
+        try (Table<Uuid, Person, ManagedResources> table =
+                     corfuStore.openTable(namespace, tableName,
+                             Uuid.class, Person.class,
+                             ManagedResources.class,
+                             // TableOptions includes option to choose - Memory/Disk based corfu table.
+                             TableOptions.fromProtoSchema(Person.class).toBuilder()
+                                     .persistentDataPath(persistedCacheLocation)
+                                     .build())) {
+
+            ManagedResources metadata = ManagedResources.newBuilder()
+                    .setCreateUser("MrProto").build();
+
+            try (TxnContext tx = corfuStore.txn(namespace)) {
+                assertThat(persons.size()).isEqualTo(ids.size());
+                Streams.zip(ids.stream(), persons.stream(), SimpleEntry::new)
+                        .forEach(pair -> tx.putRecord(table, pair.getKey(), pair.getValue(), metadata));
+                tx.commit();
+            }
+
+            try (TxnContext tx = corfuStore.txn(namespace)) {
+                final Set<Person> personSet = persons.stream()
+                        .map(person -> tx.getByIndex(table, "phoneNumber.mobile", person.getPhoneNumber().getMobile(0)))
+                        .flatMap(List::stream)
+                        .map(CorfuStoreEntry::getPayload)
+                        .collect(Collectors.toSet());
+                assertThat(persons).isEqualTo(personSet);
+                tx.txAbort();
```

Comment:
nit: abort as the last statement isn't necessary since it is the default behavior. Also we recommend callers invoke commit() for all normal transactions including read-only ones as a safeguard measure to prevent dataloss. Might be good to have the reflected in tests as well